crates/core_simd/src/ops.rs

```diff
@@ -212,7 +212,18 @@ macro_rules! impl_op {
             impl core::ops::Neg for $type {
                 type Output = Self;
                 fn neg(self) -> Self::Output {
-                    <$type>::splat(-<$scalar>::default()) - self
+                    <$type>::splat(0) - self
```

Comment:
I changed this line to {integer} 0 and the line later to -0.0 so that if you forgot or wrongly added the `@float` you'd get a compile error.